src/main/java/de/bonndan/nivio/input/InstantItemResolver.java

```diff
@@ -0,0 +1,86 @@
+package de.bonndan.nivio.input;
+
+import de.bonndan.nivio.input.dto.ItemDescription;
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.model.*;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+/**
+ * If the landscape is configured as "greedy", new items are created on the fly by a reference name.
+ *
+ * @todo add hosts as soon as relation resolver supports hosts
+ */
+public class InstantItemResolver {
+
+    private final ProcessLog log;
+
+    public InstantItemResolver(ProcessLog log) {
+        this.log = log;
+    }
+
+    public void processTargets(LandscapeDescription landscape) {
+        List<ItemDescription> all = landscape.getItemDescriptions();
+
+        if (!landscape.getConfig().isGreedy())
+            return;
```

Comment:
This should be the first thing to be called before instantiating `List<ItemDescription> all`.

Also, I personally would never omit curly braces. For guard clauses like this, it _may_ be okay, if you move the return onto the same line.

Reply:
:+1: